u2f-ref-code/java/tests/com/google/u2f/key/messages/TransferAccessMessageTest.java

```diff
@@ -0,0 +1,255 @@
+    assertEquals(VENDOR_CERTIFICATE, transferAccessMessage.getNewAttestationCertificate());
+    assertArrayEquals(TRANSFER_ACCESS_MESSAGE_SIGNATURE_USING_AUTHENTICATION_KEY_C_TO_D,
+        transferAccessMessage.getSignatureUsingAuthenticationKey());
+    assertArrayEquals(TRANSFER_ACCESS_MESSAGE_SIGNATURE_USING_ATTESTATION_KEY_C_TO_D,
+        transferAccessMessage.getSignatureUsingAttestationKey());
+  }
+
+  @Test
+  public void testTransferAccessMessage_FromBytes_ExtraBytes() {
+    try {
+      @SuppressWarnings("unused")
+      TransferAccessMessage transferAccessMessage =
+          TransferAccessMessage.fromBytes(TRANSFER_ACCESS_MESSAGE_EXTRA_BYTES);
+      fail("expected exception, but didn't get it");
+    } catch (U2FException e) {
+      assertTrue(e.getMessage().contains("Message ends with unexpected data"));
+    }
+  }
+
+  @Test
+  public void testTransferAccessMessage_FromBytes_TooFewBytes() {
+    try {
+      @SuppressWarnings("unused")
+      TransferAccessMessage transferAccessMessage =
+          TransferAccessMessage.fromBytes(TRANSFER_ACCESS_MESSAGE_TOO_FEW_BYTES);
+      fail("expected exception, but didn't get it");
+    } catch (U2FException e) {
+      assertTrue(e.getCause() instanceof java.io.EOFException);
+    }
+  }
+
+  @Test
+  public void testTransferAccessMessage_FromBytes_WayTooFewBytes() {
+    try {
+      @SuppressWarnings("unused")
+      TransferAccessMessage transferAccessMessage =
+          TransferAccessMessage.fromBytes(TRANSFER_ACCESS_MESSAGE_WAY_TOO_FEW_BYTES);
+      fail("expected exception, but didn't get it");
+    } catch (U2FException e) {
+      assertTrue(e.getCause() instanceof java.io.EOFException);
+    }
+  }
+
+  @Test
+  /* TODO(alextaka): This will break the parsing sometimes, and sometimes everything will check out.
+   * I think it just has to do with where the change gets made in the raw attestation cert. */
+  public void testTransferAccessMessage_FromBytes_BadAttestationCertificate() throws U2FException {
+    TransferAccessMessage transferAccessMessage =
+        TransferAccessMessage.fromBytes(TRANSFER_ACCESS_MESSAGE_BAD_ATTESTATION_CERT);
+    
+    assertFalse(crypto.verifySignature(transferAccessMessage.getNewAttestationCertificate(),
+        EXPECTED_TRANSFER_ACCESS_SIGNED_BYTES_FOR_ATTESTATION_KEY_A_TO_B,
+        transferAccessMessage.getSignatureUsingAttestationKey()));
+    assertTrue(crypto.verifySignature(USER_PUBLIC_KEY_SIGN,
+        EXPECTED_TRANSFER_ACCESS_SIGNED_BYTES_FOR_AUTHENTICATION_KEY_A_TO_B,
+        transferAccessMessage.getSignatureUsingAuthenticationKey()));
+  }
+
+  @Test
+  public void testTransferAccessMessage_FromBytes_CutAttestationCertificate() {
+    try {
+      @SuppressWarnings("unused")
+      TransferAccessMessage transferAccessMessage =
+          TransferAccessMessage.fromBytes(TRANSFER_ACCESS_MESSAGE_CUT_ATTESTATION_CERT);
+      // TODO(alextaka): Somehow, the length seem to match up perfectly and this returns ok.
```

Comment:
This was just dumb chance. Removing another line makes it break as expected. Changed the TestVector to cause expected behavior.